### PR TITLE
OPENTOK-41730: metal implementation as custom video driver

### PR DESCRIPTION
**Pull request by jvalli**

### CODE REVIEW

File: ARFrameMetadata.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -384,16 +425,17 @@
...
-				DEVELOPMENT_TEAM = Y6ZF2U3QJC;
+				DEVELOPMENT_TEAM = PR6C39UQ38;
 				INFOPLIST_FILE = ARFrameMetadata/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.tokbox.ARFrameMetadata;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.tokbox.ARFrameMetadata-";
```
Comment (rperezcubero) [inline ARFrameMetadata.xcodeproj/project.pbxproj:435]: I think it has a extra `-` at he end

File: ARFrameMetadata/Metal/MetalCameraSession.swift [hunk 2]
```diff
@@ -0,0 +1,294 @@
+//
+//  MetalCameraSession.swift
+//  MetalShaderCamera
+//
+//  Created by Jerónimo Valli on 9/18/20.
+//  Copyright © 2020 tokbox. All rights reserved.
+//
```
Comment (rperezcubero) [inline ARFrameMetadata/Metal/MetalCameraSession.swift:7]: We try to remove this headers from public code

File: ARFrameMetadata/OTConstants.swift [hunk 3]
```diff
@@ -8,7 +8,12 @@
 
 import Foundation
 
-let kApiKey = ""
-let kToken = ""
-let kSessionId = ""
+let kApiKey = "46888334"
```
Comment (rperezcubero) [inline ARFrameMetadata/OTConstants.swift:11]: Don't commit session information

File: ARFrameMetadata.xcodeproj/project.pbxproj (hunk 1 above)
Comment (jvalli) [reply ARFrameMetadata.xcodeproj/project.pbxproj]: Reverted!

File: ARFrameMetadata/Metal/MetalCameraSession.swift (hunk 2 above)
Comment (jvalli) [reply ARFrameMetadata/Metal/MetalCameraSession.swift]: Removed!

File: ARFrameMetadata/OTConstants.swift (hunk 3 above)
Comment (jvalli) [reply ARFrameMetadata/OTConstants.swift]: Removed!

**Comment by jvalli** [comment]
@robjperez I have updated the PR, can you take a look again when you have a chance please?

**Review by rperezcubero — APPROVED** [review]

**Comment by jvalli** [comment]
@IGitGotIt @msach22 anything else to comment? I am going to merge this PR.

**Pull request merged**